packages/core/src/createBYOKConfig.ts

```diff
@@ -0,0 +1,80 @@
+import invariant from 'tiny-invariant'
+import type { SignMessageReturnType } from 'viem'
+import type { BaseConfig } from './createConfig.js'
+import type * as rustUtils from './utils.d.ts'
+
+export type CreateBYOKConfigParameters = {
```

Comment:
I don't think BYOK (bring your own key) is a good name externally. it was fine as an internal label to discuss this effort, but we should rename this to `ExternallyManagedKey`, `ExternalKey` or something like that